aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -23,6 +23,7 @@ DeviceType SparseCsrTensorSetToDeviceType(DispatchKeySet key_set) {
 
 SparseCsrTensorImpl::SparseCsrTensorImpl(
     at::DispatchKeySet key_set,
+    Layout layout,
```

Comment:
Why do you have to take this as an argument, the layout is guaranteed to be CSR here no?

Reply:
Answer: because it can also be BSR. Consider adding an assert here.

Reply:
Maybe due for a rename of SparseCsrTensorImpl itself

Reply:
No, with this PR, SparseCsrTensorImpl can hold four layouts: kSparseCsr, kSparseCsc, kSparseBsr, and kSparseBsc

Reply:
Renaming SparseCsrTensorImpl makes a lot of sense. However, since this PR implements a new approach (dispatch key-layout relation is not many-to-one anymore), I'd suggest keeping the PR minimal for reviewers and doing the rename in a follow-up PR (which would be simple but very verbose).